prompt.go

```diff
@@ -70,12 +79,17 @@ func (p *Prompt) Run() {
 
 				// Unset raw mode
 				// Reset to Blocking mode because returned EAGAIN when still set non-blocking mode.
+				p.inNotTearedDown = false
 				debug.AssertNoError(p.in.TearDown())
 				p.executor(e.input)
 
 				p.completion.Update(*p.buf.Document())
+
 				p.renderer.Render(p.buf, p.completion)
 
+				if p.exitor != nil && p.exitor(e.input) {
+					return
```

Comment:
You can simplify by puttinng `p.inNotTearedDown = false` here because you can remove L67 and L82.

Furtheremore, I have two question about the name of `inNotTearedDown`.

* Is it typo of `inNotTearDown` ?
* I couldn't understand why you call `p.in.TearDown()` when `inNotTearedDown = true`. From the meaning of in**Not**TearedDown, I guess you want to avoid calling `p.in.TearDown()` when it's true. Right?

Reply:
@c-bata the idea is to call `p.in.TearDown()`  is not *already* teared down (hence `inNotTearedDown`, but `inNotTearDown` would work too)
The defer function should call `p.in.TearDown()` only if it was *not* already done before (or you get a panic: "close of closed channel" initially described in https://github.com/c-bata/go-prompt/pull/161/commits/a7d14f3e07021ac8c419c18b66f0c0cb8ba000ad.

I can remove L67 and L82.
Do you want me to rename `inNotTearedDown` to `inNotTearDown`? Or `inNotAlreadyTearDown`?
Or maybe there is a better name :)

Reply:
Oh, thank you for your explanation. I misunderstood.

I prefer:

1. Rename `inNotTearedDown ` to `skipTearDown` (default: `false`).
1. Set `p.skipTearDown = true` at L91.
1. Skip calling `p.in.TearDown()` if `p.skipTearDown` is true.

Reply:
@c-bata Done. 
I guess I named it initially `inNotxxx` in order to test for true in L285, instead of false.

Note: L45 is technically not needed (the zero value for a `bool` is `false`), but... just to be on the safe side, I initialize `skipTearDown` to `false` anyway.